From b25fd7980073dde6baa11bb36d9e61e069764ce2 Mon Sep 17 00:00:00 2001
From: Ethan Halsall <ethan.halsall@gmail.com>
Date: Thu, 9 Apr 2020 23:35:54 +0100
Subject: [PATCH] SystemUI: Do not use new lockscreen layout for bypass

bypassEnabled will always evaluate to false now.
Introduce bypassEnabledBiometric which is only used to tell the biometric service whether we can bypass lockscreen without swipe, dependant on Settings.Secure.FACE_UNLOCK_DISMISSES_KEYGUARD.

This enables full lockscreen layout when "Skip lockscreen" setting is enabled, while also fixing status bar glitches.

Change-Id: Ibdbf16df18543382219a75ce1aaf4e25f3fb4ddb
Signed-off-by: Adithya R <gh0strider.2k18.reborn@gmail.com>
Signed-off-by: Mohammad Hasan Keramat J <ikeramat@protonmail.com>
---
 .../statusbar/phone/BiometricUnlockController.java    |  2 +-
 .../statusbar/phone/KeyguardBypassController.kt       | 11 +++++++----
 2 files changed, 8 insertions(+), 5 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/BiometricUnlockController.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/BiometricUnlockController.java
index 318d35ec3bd3..6c4a7b3a409a 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/BiometricUnlockController.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/BiometricUnlockController.java
@@ -686,7 +686,7 @@ public class BiometricUnlockController extends KeyguardUpdateMonitorCallback imp
         final boolean unlockingAllowed =
                 mUpdateMonitor.isUnlockingWithBiometricAllowed(isStrongBiometric);
         final boolean deviceDreaming = mUpdateMonitor.isDreaming();
-        final boolean bypass = mKeyguardBypassController.getBypassEnabled()
+        final boolean bypass = mKeyguardBypassController.getBypassEnabledBiometric()
                 || mAuthController.isUdfpsFingerDown();
         final boolean isBouncerShowing = mKeyguardViewController.primaryBouncerIsOrWillBeShowing()
                 || mKeyguardTransitionInteractor.getCurrentState()
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/KeyguardBypassController.kt b/packages/SystemUI/src/com/android/systemui/statusbar/phone/KeyguardBypassController.kt
index cd8977b26e2a..3deb4f90ed47 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/KeyguardBypassController.kt
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/KeyguardBypassController.kt
@@ -118,6 +118,8 @@ constructor(
             notifyListeners()
         }
 
+    var bypassEnabledBiometric: Boolean = false
+
     var bouncerShowing: Boolean = false
     var launchingAffordance: Boolean = false
     var qsExpanded = false
@@ -155,7 +157,7 @@ constructor(
                     1
                 else 0
             tunerService.addTunable(
-                { key, _ -> bypassEnabled = tunerService.getValue(key, dismissByDefault) != 0 },
+                { key, _ -> bypassEnabledBiometric = tunerService.getValue(key, dismissByDefault) != 0 },
                 Settings.Secure.FACE_UNLOCK_DISMISSES_KEYGUARD,
             )
             lockscreenUserManager.addUserChangedListener(
@@ -198,8 +200,8 @@ constructor(
         biometricSourceType: BiometricSourceType,
         isStrongBiometric: Boolean,
     ): Boolean {
-        if (biometricSourceType == BiometricSourceType.FACE && bypassEnabled) {
-            val can = canBypass()
+        if (bypassEnabledBiometric) {
+            val can = biometricSourceType != BiometricSourceType.FACE || canBypass()
             if (!can && (isPulseExpanding || qsExpanded)) {
                 pendingUnlock = PendingUnlock(biometricSourceType, isStrongBiometric)
             }
@@ -227,7 +229,7 @@ constructor(
 
     /** If keyguard can be dismissed because of bypass. */
     fun canBypass(): Boolean {
-        if (bypassEnabled) {
+        if (bypassEnabledBiometric) {
             return when {
                 bouncerShowing -> true
                 keyguardTransitionInteractor.getCurrentState() == KeyguardState.ALTERNATE_BOUNCER ->
@@ -261,6 +263,7 @@ constructor(
             pw.println("  mPendingUnlock: $pendingUnlock")
         }
         pw.println("  bypassEnabled: $bypassEnabled")
+        pw.println("  bypassEnabledBiometric: $bypassEnabledBiometric")
         pw.println("  canBypass: ${canBypass()}")
         pw.println("  bouncerShowing: $bouncerShowing")
         pw.println(
-- 
2.49.0

